pkg/cloudprovider/cloudprovider.go

```diff
@@ -103,11 +112,26 @@ func New(ctx awscontext.Context) *CloudProvider {
 				lo.Must(getCABundle(ctx.RESTConfig)),
 				ctx.StartAsync,
 				kubeDNSIP,
+				clusterEndpoint,
 			),
 		),
 	}
 }
 
+func resolveClusterEndpoint(ctx context.Context, eksAPI eksiface.EKSAPI) (string, error) {
+	resolveClusterEndpointFromSettings := settings.FromContext(ctx).ClusterEndpoint
+	if resolveClusterEndpointFromSettings != "" {
+		return resolveClusterEndpointFromSettings, nil // cluster endpoint is explicitly set
+	}
+	out, err := eksAPI.DescribeCluster(&eks.DescribeClusterInput{
```

Comment:
Can you add this new permission (eks:DescribeCluster) here https://github.com/aws/karpenter/blob/main/website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml#L43

And also put a note in the upgrade guide for version v1.25.0 that an additional permissions is required? https://github.com/aws/karpenter/blob/main/website/content/en/preview/upgrade-guide.md#released-upgrade-notes